Dockerfile

```diff
@@ -1,13 +1,15 @@
-FROM golang@sha256:a92abf53d0ac8dd292b432525c15ce627554546ebc4164d1d4149256998122ce
+# docker.io/golang:1.17-alpine
+FROM golang@sha256:8474650232fca6807a8567151ee0a6bd2a54ea28cfc93f7824b42267ef4af693
 
-RUN apk add make
+RUN apk add make git
 
 ADD go.mod go.mod
 ADD go.sum go.sum
 
 ENV GOPATH=""
 ENV CGO_ENABLED=0
-RUN go mod download
+
+RUN go mod tidy -compat=1.17
```

Comment:
I'm replacing `go mod download` with this because the former causes this error since upgrading to go 1.17: 
```
go mod tidy: error loading go 1.16 module graph: github.com/hashicorp/vault@v1.9.2 requires
        github.com/hashicorp/vault/api/auth/approle@v0.0.0-00010101000000-000000000000: invalid version: unknown revision 000000000000
```
which I can reproduce when running `go mod tidy` without the `-compat=1.17` flag. As it turns out `go mod tidy` will not only delete unneeded dependencies, but download the ones strictly needed for the main package according to https://github.com/golang/go/issues/43994#issuecomment-770053099. So it seems that is enough to get what we want.